### PR TITLE
Order details screen: add option to delete shipment tracking

### LINKED ISSUE #876: Order details screen: add option to delete shipment tracking
As discussed [in this p2 post](https://wp.me/p99K0U-19M-p2), we should add the ability to delete each shipment tracking. Deleting and opening a web view to track the shipment should be grouped.

woo-mvp-android-ios-i8-20190218-copy-2019-04-11-17-20-48

### PR DESCRIPTION
**Pull request by ctarda**
Fixes #876 

## Changes
- Updated `OrderTrackingTableViewCell` to remove the "track" button and a separator
- Updated `OrderDetailsViewController` to present an alert sheet and to add the delete action

## Testing
- Checkout the branch, run the tests
- Navigate to an order with shipment trackings (`OrderDetailsViewController`)
- Tap one of the trackings. Tap the Delete Tracking and Track Shipment buttons and check that they work as expected

**Note**: When presenting the sheet in the simulator, on iPhone, there will be an Autolayout error:

```
2019-05-14 09:14:16.959312+0800 WooCommerce[12158:259530] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x60000164e170 UIView:0x7fab0565b430.width == - 16   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x60000164e170 UIView:0x7fab0565b430.width == - 16   (active)>
```

We have discussed that error in the past, and found it to be out of our control (p1556162547026400-slack-woo-mobile)

- [ ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

**Comment by mindgraffiti** [comment]
A neat trick for keeping your animated .gifs down to a manageable screen size is to take a GitHub embed link:

```
![Simulator Screen Shot - iPhone Xs - 2019-05-07 at 16 26 38](https://user-images.githubusercontent.com/1062444/57713997-2a83b980-7639-11e9-86fa-89258f3629ec.png)

```

And pull out the image link. Then use plain HTML to force it to size:
```
<img src="https://user-images.githubusercontent.com/1062444/57713997-2a83b980-7639-11e9-86fa-89258f3629ec.png" width="350" />
```

File: WooCommerce/Classes/ViewRelated/Orders/OrderDetailsViewController.swift [hunk 1]
```diff
@@ -666,6 +654,27 @@ private extension OrderDetailsViewController {
...
+    func deleteTracking(_ tracking: ShipmentTracking) {
+        let siteID = viewModel.order.siteID
+        let orderID = viewModel.order.orderID
+        let trackingID = tracking.trackingID
+
+        let deleteTrackingAction = ShipmentAction.deleteTracking(siteID: siteID,
+                                                                 orderID: orderID,
+                                                                 trackingID: trackingID) { [weak self] error in
+                                                                    if let error = error {
+                                                                        DDLogError("⛔️ Delete Tracking Failure: orderID \(orderID). Error: \(error)")
```
Comment (mindgraffiti) [inline WooCommerce/Classes/ViewRelated/Orders/OrderDetailsViewController.swift]: When creating error messages, it also helps to say where the error is from. So I would say "⛔️ Order Details - Delete Tracking Failure: orderID \(orderID). Error: \(error)" to make it easier when troubleshooting issues from a emailed Error log. If you find hints like that useful, think about adding it here too.

File: WooCommerce/Classes/ViewRelated/Orders/OrderDetailsViewController.swift [hunk 2]
```diff
@@ -1093,6 +1166,12 @@ private extension OrderDetailsViewController {
         static let message = NSLocalizedString("Message", comment: "Message phone number button title")
     }
 
+    enum TrackingAction {
+        static let dismiss = NSLocalizedString("Dismiss", comment: "Dismiss the shipment tracking action sheet")
+        static let trackShipment = NSLocalizedString("Track Shipment", comment: "Track shiopment button title")
```
Comment (mindgraffiti) [inline WooCommerce/Classes/ViewRelated/Orders/OrderDetailsViewController.swift]: `shipment` instead of `shiopment`

**Comment by mindgraffiti** [comment]
Please edit your PR comment to use the `Closes` or `Fixes` keyword in front of the GitHub ticket number. It will make the ticket number automatically close when you merging this PR.

**Comment by mindgraffiti** [comment]
Comment test. Ref 400&cid=C6H8C3G23-slack-C6H8C3G23

**Comment by ctarda** [comment]
Thanks @mindgraffiti I just pushed an update that adds the missing ellipsis. Also, it will mimic the behaviour of the cell in the fulfillment screen, where the sheet will be presented only when tapping the button, but not when tapping anywhere else in the cell.

**Comment by mindgraffiti** [comment]
Hey Cesar! This is so close. I apologize, I should have been more specific about the interactions for the row. When we have a single interaction on a row (tap a button), this is what we build:

1. Disable the button interaction and selection (or make the accessoryView a simple imageView)
2. For the cell, turn on Selection: default by either adding it to the cell configuration code `cell.selectionStyle = .default` or by editing the nib file in Interface builder (the dropdown menu that says "Selection" and select "default"
3. In `didSelectRow:` initiate the action (present the action sheet) on that row.

(Same thing you already did for the "Add Tracking" row.)

@kyleaparker please correct me if I'm wrong about our interaction standards for iOS ☝️ 

After completing those, :shipit:

**Comment by ctarda** [comment]
@mindgraffiti So do we need to update the cells in the FulfillViewController to behave the same way? At the moment, when you want to delete a tracking in the FulfillViewController, only the accessory view (the delete button) is tappable and the cell is not selectable.

**Comment by mindgraffiti** [comment]
That's a good point. For any positive interactions we make the whole row tappable. Since the more button is similar to the pencil button (an edit operation) or a delete (a destructive operation), then you're right. It should stay the way it is. You can ignore my earlier comment 🙂

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by ctarda** [comment]
Thanks for clarifying, I think I've got it now! 🙇

**Comment by astralbodies** [comment]
If it's supposed to be like the (i) info disclosure indicator I *think* the row shouldn't be tappable.

**Comment by astralbodies** [comment]
lol stupid GitHub didn't update the conversation before I added the comment. Ignoreeeee meeeee

**Pull request merged**